src/PrestaShopBundle/Controller/Admin/Improve/Modules/AddonsStoreController.php

```diff
@@ -35,28 +36,37 @@
 class AddonsStoreController extends FrameworkBundleAdminController
 {
     /**
-     * @var string the controller name for routing
-     */
-    const CONTROLLER_NAME = 'AdminAddonsCatalog';
-
-    /**
+     *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
      * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
     public function indexAction(Request $request)
     {
+        $pageContent = @file_get_contents($this->getAddonsUrl($request));
+
+        if (!$pageContent) {
+            $this->addFlash('error', $this->trans(
+                'An unexpected error occurred.',
```

Comment:
Do you think that it will be interesting here to bring an element of solution in this wording? Our point, for error notifications, is to be less formal and more helpful to the users.

Reply:
@LouiseBonnard Good idea 👍 do you have suggestions ? This error message will be displayed if for any reason the shop is unable to get the data to be displayed from Addons marketplace. It could be a network issue, could be a temporary Addons failure, could be a firewall issue ...

Basically the message is "Sorry we are unable to display this page because we cannot reach Addons." . How would you turn that ?

I'll do it in another PR though as mickaelandrieu requested it.

Reply:
Other PR: https://github.com/PrestaShop/PrestaShop/pull/12699